eng/Versions.props

```diff
@@ -18,7 +18,7 @@
     <!-- Opt-in/out repo features -->
     <UsingToolMicrosoftNetCompilers>true</UsingToolMicrosoftNetCompilers>
     <!-- TODO: Upgrade compiler version to enable Static Abstracts in Interfaces, interpolated string handlers, and caller arg expressions; remove this once the employed SDK uses a new enough version. -->
```

Comment:
Does this resolve the TODO, or do we have more upgrades to do?

Reply:
No, we need to figure out why the compiler we're getting without this is really old:
https://github.com/dotnet/runtime/issues/57583
fix that, and then delete both this TODO and the pinning of the version.

In the meantime, I'm bumping it up to something newer.